src/mgr/BaseMgrModule.cc

```diff
@@ -404,19 +404,35 @@ ceph_get_module_option_ex(BaseMgrModule *self, PyObject *args)
 static PyObject*
 ceph_get_module_option(BaseMgrModule *self, PyObject *args)
 {
-  char *key = nullptr;
-  if (!PyArg_ParseTuple(args, "s:ceph_get_module_option", &key)) {
+  // Do not call ceph_get_module_option_ex() here, otherwise
+  // it will end up in a dead-lock in some situations.
+  // Call stack:
+  // PyModuleRegistry::active_start()
+  // ActivePyModules::start_one()
+  // ActivePyModule::load()
+  // ...
+  // ActivePyModules::get_typed_config()
+  // PyModuleRegistry::module_exists()
+  char *what = nullptr;
```

Comment:
i like `key` more. as it matches the parameter name being used elsewhere. please `git grep ceph_get_module_option` for more examples.

Reply:
That's the original code, but you're right, ``key`` is much better. Will adapt this.